cli/azd/cmd/deps_record.go

```diff
@@ -19,6 +20,17 @@ func createHttpClient() *http.Client {
 	transport := http.DefaultTransport.(*http.Transport).Clone()
 	// Allow for self-signed certificates, which is what the recording proxy uses.
 	transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	// AZD_TEST_HTTPS_PROXY is the proxy setting that only affects azd in record mode.
+	// This is useful since the recording proxy server isn't trusted by other processes currently.
```

Comment:
Just so I understand - this is because the HTTPS proxy has to use a self-signed certificate so it can MITM the traffic and so clients which validate the chain of the server certificate fail, and if we set HTTPS_PROXY directly it would mean subprograms might pick this up and try to use it and fail?

Reply:
Correct. We could also use add the certificate to the trust chain on a devbox which the subprograms would trust it as well, and this would all work as expected. But the proxy is a true MITM, so it's a full-blown trust for all domains.

I was really torn about making a decision here for the longest time, but @wbreza hit this again recently and I think our way forward is to limit HTTP proxying to `azd`. For subprograms we intend to "mock out", we would use command recording -- I think this ends up being the best tradeoff, we don't want to necessarily dive deeper into understanding requests made by subprograms (although it is really cool that this all works by default if we wanted).